### PR TITLE
config(ticdc): add ticdc labels

### PR DESCRIPTION
**Pull request by hi-rustin**
1. Updated and added descriptions.
2. Removed `contribution-welcome label `due to duplication with `status/help-wanted`. Also, change the `status/help-wanted` and `good-first-issue` to the standard GitHub-identified format of
`help wanted` and `good first issue`.
3. Delete `needs-cherry-pick-release-5.0-rc` because we don't use it anymore.
4. Rename `ptal` to `status/ptal`.

[rendered](https://github.com/hi-rustin/configs/blob/rustin-patch-ticdc/prow/config/labels.md)

### CODE REVIEW

**Review by overvenus — APPROVED** [review]
LGTM!

PTAL @amyangfei

File: prow/config/labels.md [hunk 1]
```diff
@@ -510,6 +513,80 @@ larger set of contributors to apply/remove them.
...
+| <a id="status/need-discussion" href="#status/need-discussion">`status/need-discussion`</a> | Issue that needs to be discussed to confirm priority, milestone, plan and task breakdown.| anyone |  [ti-community-label](https://book.prow.tidb.io/#/en/plugins) |
+| <a id="type/bug" href="#type/bug">`type/bug`</a> | This is a bug.| anyone |  [ti-community-label](https://book.prow.tidb.io/#/en/plugins) |
+| <a id="wontfix" href="#wontfix">`wontfix`</a> | This issue will not be fixed.| anyone |  [ti-community-label](https://book.prow.tidb.io/#/en/plugins) |
+
+## Labels that apply to pingcap/ticdc, only for PRs
+
+| Name | Description | Added By | Prow Plugin |
+| ---- | ----------- | -------- | --- |
+| <a id="cherry-pick-approved" href="#cherry-pick-approved">`cherry-pick-approved`</a> | The PR of cherry-pick was approved.| lonng | |
+| <a id="component/dependencies" href="#component/dependencies">`component/dependencies`</a> | Pull requests that update a dependency fil.| anyone |  [ti-community-label](https://book.prow.tidb.io/#/en/plugins) |
```
Comment (amyangfei) [inline prow/config/labels.md:576]:
```suggestion
| <a id="component/dependencies" href="#component/dependencies">`component/dependencies`</a> | Pull requests that update a dependency file.| anyone |  [ti-community-label](https://book.prow.tidb.io/#/en/plugins) |
```

File: prow/config/external_plugins_config.yaml [hunk 2]
```diff
@@ -555,26 +555,17 @@ ti-community-label:
...
-      - 'bug-from-user'
-      - 'contribution-welcome'
-      - 'dba-test'
-      - 'dependencies'
-      - 'documentation'
       - 'duplicate'
-      - 'enhancement'
-      - 'feature-request'
-      - 'good-first-issue'
       - 'hacktoberfest'
```
Comment (amyangfei) [inline prow/config/external_plugins_config.yaml:559]: Should we keep this, it's a community activity, which has ended. (In fact no one took part in.

**Comment by amyangfei** [comment]
A small question, will `status/ptal` and `status/LGTx` (where x can be 1, 2 or 3) are mutually exclusive?

File: prow/config/labels.md (hunk 1 above)
Comment (hi-rustin) [reply prow/config/labels.md]: Good catch! Fixed.

File: prow/config/external_plugins_config.yaml (hunk 2 above)
Comment (hi-rustin) [reply prow/config/external_plugins_config.yaml]: Deleted.

**Review by amyangfei — APPROVED** [review]

**Comment by hi-rustin** [comment]
> A small question, will `status/ptal` and `status/LGTx` (where x can be 1, 2 or 3) are mutually exclusive?

 I don't think it affects it because we generally require multiple stages of review, plus we can find the current PR from the links in the bot's replies that require several LGTM. So I don't think adding a prefix will have much of an impact.

**Pull request merged**